### DISCUSSION TITLE
ubuntu22.04 High-DPI  blurry

### DISCUSSION THREAD

**Discussion by qfmee** [AvaloniaUI/Avalonia#13450] [General]
The display will become blurry in ubuntu22.04 High-DPI mode，Does avalonia support the Wayland protocol?

**Comment by thevortexcloud** [comment]
Avalonia does not have native support for Wayland. There is [this](https://github.com/AvaloniaUI/Avalonia/pull/11546) WIP PR to add a native Wayland backend though. But at the moment everything goes via XWayland.

My guess though is you have fractional scaling enabled for your DE. Which can cause issues in general on Linux for a lot of apps. Try using an integer value.

**Reply by qfmee** [reply]
200% will also blur, how to solve the problem of display blur with high resolution! Is there a solution?

**Reply by thevortexcloud** [reply]
Does it look fine at 100% scaling? If so that's not really Avalonia's problem. Avalonia does have ways to set it's own scaling though which _may_ help. See this issue: https://github.com/AvaloniaUI/Avalonia/issues/4826

Scaling on Linux is generally a mess. Gnome's implementation is fairly weird too. Some of the options listed [here](https://wiki.archlinux.org/title/HiDPI) may help or switching to X11/different DE such as KDE.
